frontend/src/js/external-forms/form-concept-group/FormConceptGroup.tsx

```diff
@@ -396,9 +453,30 @@ const FormConceptGroup = (props: Props) => {
             );
           }}
           onDropConcept={(concept) => {
-            const { valueIdx, conceptIdx } = editedFormQueryNodePosition;
+            let { valueIdx, conceptIdx } = editedFormQueryNodePosition;
+            let updatedValue = props.value;
+            if (isMovedObject(concept)) {
+              const { movedFromFieldName, movedFromAndIdx, movedFromOrIdx } =
+                concept.dragContext;
+              valueIdx = valueIdx > movedFromAndIdx ? valueIdx - 1 : valueIdx;
+              if (movedFromFieldName === props.fieldName) {
+                updatedValue =
+                  updatedValue[movedFromAndIdx].concepts.length === 1
+                    ? removeValue(updatedValue, movedFromAndIdx)
+                    : removeConcept(
+                        updatedValue,
+                        movedFromAndIdx,
+                        movedFromOrIdx,
+                      );
+                setEditedFormQueryNodePosition({ valueIdx, conceptIdx });
```

Comment:
Could you explain this part – why is it required to set a potentially different node as the one to be edited?

Reply:
When we delete a concept which is above the currently selected concept (in the Concept editing menu) then the index will change -> on rerender a different concept will be shown in the modal.

=> I extracted a variable and added a clarifying comment in the code

Reply:
Aah I see, thanks for clarifying